src/utils.py

```diff
@@ -21,7 +23,10 @@ def paginate(iterable: Iterable[T], page: int, limit: int, request: Request) ->
     prev_page = None
     next_page = None
     data = []
-    for idx, item in enumerate(iterable):
+    iter_data, iter_total = itertools.tee(iterable)
+    total_items = functools.reduce(lambda prev, curr: prev + 1, iter_total, 0)
+    total_pages = round(total_items / limit)
```

Comment:
I think you want `math.ceil` here instead of `round`. If you have `total_items = 3` and `limit = 10`, `total_pages` will be `0` if you use `round`. Otherwise this looks good!

Reply:
Thank you, Patrick. I'll fix that.